File: Algorithms/Java/209-Minimum_Size_Subarray_Sum.java
// Runtime: 1 ms, faster than 99.97% of Java online submissions for Minimum Size Subarray Sum.
// Memory Usage: 38.9 MB, less than 65.83% of Java online submissions for Minimum Size Subarray Sum.
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int res = Integer.MAX_VALUE;
        int start = 0;
        int sum = 0;
        for (int end = 0; end < nums.length; ++end) {
            sum += nums[end];
            while (sum >= target) {
                res = Math.min(res, end + 1 - start);
                sum -= nums[start];
                start++;
            }
        }
        return res != Integer.MAX_VALUE ? res : 0;
    }
}

File: Algorithms/C++/209-Minimum_Size_Subarray_Sum.cpp
// Runtime: 4 ms, faster than 98.83% of C++ online submissions for Minimum Size Subarray Sum.
// Memory Usage: 10.7 MB, less than 53.15% of C++ online submissions for Minimum Size Subarray Sum.

#include <climits>

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int res = INT_MAX;
        int left = 0;
        int sum = 0;
        for (int right = 0; right < nums.size(); ++right) {
            sum += nums[right];
            while (sum >= target) {
                res = min(res, right + 1 - left);
                sum -= nums[left];
                left++;
            }
        }
        return res != INT_MAX ? res : 0;
    }
};
